solr/core/src/test/org/apache/solr/cli/TestSolrCLIRunExample.java

```diff
@@ -325,6 +330,99 @@ public void testFilmsExample() throws Exception {
     testExample("films");
   }
 
+  @Test
+  public void testAdditionalArgs() throws Exception {
+    String testStandaloneMode = LuceneTestCase.rarely() ? "--user-managed" : "";
+    Path defaultSolrHomeDir = ExternalPaths.SERVER_HOME;
+    if (!Files.isDirectory(defaultSolrHomeDir)) {
+      fail(defaultSolrHomeDir + " not found and is required to run this test!");
+    }
+
+    Path testSolrHomeDir = createTempDir();
+    Path solrServerDir = defaultSolrHomeDir.getParent();
+
+    // need a port to start the example server on
+    int bindPort = -1;
+    try (ServerSocket socket = new ServerSocket(0)) {
+      bindPort = socket.getLocalPort();
+    }
+
+    log.info("Selected port {} to start techproducts example Solr instance on ...", bindPort);
+
+    String sysProp = "solr.customprop.greet";
+    String[] toolArgs =
+        new String[] {
+          "-e",
+          "techproducts",
+          "--server-dir",
+          solrServerDir.toString(),
+          "--solr-home",
+          testSolrHomeDir.toString(),
+          "-p",
+          String.valueOf(bindPort),
+          testStandaloneMode,
+          "--jvm-opts",
+          "-D" + sysProp + "=hello"
+        };
+
+    // capture tool output to stdout
+    CLITestHelper.TestingRuntime runtime = new CLITestHelper.TestingRuntime(true);
+
+    RunExampleExecutor executor = new RunExampleExecutor();
+    closeables.add(executor);
+
+    RunExampleTool tool = new RunExampleTool(executor, System.in, runtime);
```

Comment:
I wonder if a helper method would reduice duplication between methods around the running othe tool and hanlding of the sockets etc?